sites/public/src/pages/account/application/[id]/lottery-results.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useState, useContext } from "react"
+import { useRouter } from "next/router"
+import { t } from "@bloom-housing/ui-components"
+import { AuthContext, BloomCard, CustomIconMap, RequireLogin } from "@bloom-housing/shared-helpers"
+import { Application, Listing } from "@bloom-housing/shared-helpers/src/types/backend-swagger"
+import { Card, Button, Heading, Icon, Message } from "@bloom-housing/ui-seeds"
+import FormsLayout from "../../../../layouts/forms"
+import {
+  ApplicationError,
+  ApplicationListingCard,
+} from "../../../../components/account/ApplicationCards"
+import styles from "../../../../../styles/lottery-results.module.scss"
+
+export default () => {
+  const router = useRouter()
+  const applicationId = router.query.id as string
+  const { applicationsService, listingsService, profile } = useContext(AuthContext)
+  const [application, setApplication] = useState<Application>()
+  const [listing, setListing] = useState<Listing>()
+  const [unauthorized, setUnauthorized] = useState(false)
+  const [noApplication, setNoApplication] = useState(false)
+  useEffect(() => {
+    if (profile) {
+      applicationsService
+        .retrieve({ applicationId })
+        .then((app) => {
+          setApplication(app)
+          listingsService
+            ?.retrieve({ id: app.listings.id })
+            .then((retrievedListing) => {
+              setListing(retrievedListing)
+            })
+            .catch((err) => {
+              console.error(`Error fetching listing: ${err}`)
+            })
```

Comment:
Taking a look at the network calls, it appears like the listing name and id are returned in the application retrieve call. So this second call is unnecessary unless you need more than just those two fields

Reply:
Ah yeah this was copy paste from the application page - we will also need the number of units as part of the BE ticket here.

Reply:
Good call out. I'm cool with refactoring this when the backend endpoint is flushed out!